server/src/main/java/uk/ac/cam/cl/echo/extrusionfinder/server/sourcer/crawlers/ExtendedCrawler.java

```diff
@@ -0,0 +1,31 @@
+package uk.ac.cam.cl.echo.extrusionfinder.server.sourcer.crawlers;
+
+import uk.ac.cam.cl.echo.extrusionfinder.server.parts.Part;
+import edu.uci.ics.crawler4j.crawler.WebCrawler;
+import java.util.List;
+
+/**
+ * Extends the crawler4j web crawler with, essentially, dynamic information
+ * injection for use in a static context ...
+ */
+public abstract class ExtendedCrawler extends WebCrawler {
+
+    /**
+     * @param parts The stream builder that extrusion parts are added to.
+     * Should update subclass with the parts stream such that the subclass puts
+     * its results in the stream. Kind of hacky, but necessary due to the
+     * staticness of the crawler4j implementation.
+     */
+    public abstract void configure(List<Part> parts);
+
+    /**
+     * @return  URL seeds for this crawler.
+     */
+    public abstract String[] getSeeds();
```

Comment:
Since this is an abstract class, why not define SEEDS and MANUFACTURER_ID in this class?